File: src/unwrittenfun/minecraft/unwrittenblocks/gui/slots/SlotGpsChip.scala
package unwrittenfun.minecraft.unwrittenblocks.gui.slots

import net.minecraft.inventory.{IInventory, Slot}
import net.minecraft.item.ItemStack
import unwrittenfun.minecraft.unwrittenblocks.items.UBItems

/**
 * Mod: UnwrittenBlocks
 * Author: UnwrittenFun
 * License: Minecraft Mod Public License (Version 1.0.1)
 */
class SlotGpsChip(inventory: IInventory, id: Int, x: Int, y: Int) extends Slot(inventory, id, x, y) {
  override def isItemValid(stack: ItemStack): Boolean = stack.itemID == UBItems.gpsChip.itemID && stack.getItemDamage == 1

  override def getSlotStackLimit: Int = 1
}

File: WallTeleporters/src/unwrittenfun/minecraft/wallteleporters/gui/slots/SlotGpsChip.java
package unwrittenfun.minecraft.wallteleporters.gui.slots;

import net.minecraft.inventory.IInventory;
import net.minecraft.inventory.Slot;
import net.minecraft.item.ItemStack;
import unwrittenfun.minecraft.wallteleporters.items.WTItems;

/**
 * Mod: Wall Teleporters
 * Author: UnwrittenFun
 * License: Minecraft Mod Public License (Version 1.0.1)
 */
public class SlotGpsChip extends Slot {
    public SlotGpsChip(IInventory inventory, int id, int x, int y) {
        super(inventory, id, x, y);
    }

    @Override
    public boolean isItemValid(ItemStack stack) {
        return stack.itemID == WTItems.gpsChip.itemID && stack.getItemDamage() == 1;
    }

    @Override
    public int getSlotStackLimit() {
        return 1;
    }
}
